src/Clapi/TextSerialisation.hs

```diff
@@ -46,12 +46,17 @@ msgBuilder msg = case msg of
     noSubs _ = []
     valB methodChar subs msg = fromChar methodChar <> tab subs msg
 
+headerBuilder :: Message -> Builder
+headerBuilder m = fromString $ fmap typeTag $ msgArgs' m
+
+-- Not sure how useful this is because how often do you know all the messages
+-- up front?
```

Comment:
Does Haskell's laziness buy us anything here? It could conceptually be the infinite list of messages coming in and the `Builder` could be streaming them out the other end?

Reply:
Not without unsafePerformIO unless I'm missing something, you can generate an infinite list from a pattern, but not from input.